build-release.sh

```diff
@@ -23,50 +23,68 @@ sign_windows() {
 }
 
 sign_macos() {
-  if [ ! -z "${OSX_HOST}" ]; then
-    osx_tmpdir=$(ssh "${OSX_HOST}" "mktemp -d")
-    reldir="$1"
-    binname="$2"
-    is_mono="$3"
-
-    if [[ "${is_mono}" == "1" ]]; then
-      appname="Godot_mono.app"
-      entitlements=editor_mono.entitlements
-      sharpdir="${appname}/Contents/Resources/GodotSharp"
-      extra_files="${sharpdir}/Mono/lib/*.dylib ${sharpdir}/Tools/aot-compilers/*/*"
-    else
-      appname="Godot.app"
-      entitlements=editor.entitlements
-    fi
-
-    scp "${reldir}/${binname}.zip" "${OSX_HOST}:${osx_tmpdir}"
-    scp "${basedir}/build-macosx/${entitlements}" "${OSX_HOST}:${osx_tmpdir}"
+  if [ -z "${OSX_HOST}" ]; then
+    return
+  fi
+  osx_tmpdir=$(ssh "${OSX_HOST}" "mktemp -d")
+  reldir="$1"
+  binname="$2"
+  is_mono="$3"
+
+  if [[ "${is_mono}" == "1" ]]; then
+    appname="Godot_mono.app"
+    entitlements=editor_mono.entitlements
+    sharpdir="${appname}/Contents/Resources/GodotSharp"
+    extra_files="${sharpdir}/Mono/lib/*.dylib ${sharpdir}/Tools/aot-compilers/*/*"
```

Comment:
For export templates this file list is different:

```
    sharpdir="${appname}/Contents/Resources/"
    extra_files="${sharpdir}/data.mono.osx.64.release/Mono/lib/*.dylib ${sharpdir}/data.mono.osx.64.release_debug/Mono/lib/*.dylib"
```

Reply:
That's actually the editor signing routine here. I split the cosmetic changes in a separate commit now to avoid confusion.

But if we need to sign the Mono files too for export templates, then I should re-add similar logic to `code_sign_template()`.

Reply:
Yes, templates also need `dylib`s to be signed. This should go to the `sign_macos_template()`.